File: ch04/19/Mirror.cpp
// 操作给定的二叉树，将其变换为源二叉树的镜像
/*
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};*/
class Solution {
public:
    //第二遍刷题
    void Mirror(TreeNode *pRoot) {
        if (pRoot == NULL)
            return;
        TreeNode *tmpNode = pRoot->left;
        pRoot->left = pRoot->right;
        pRoot->right = tmpNode;
        Mirror(pRoot->left);
        Mirror(pRoot->right);
    }
};

File: ch04/19/Mirror.py
# -*- coding:utf-8 -*-
# 操作给定的二叉树，将其变换为源二叉树的镜像
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回镜像树的根节点
    def Mirror(self, root):
        # write code here
        if root == None:
            return None
    
        p = root
        tmp = p.left
        p.left = p.right
        p.right = tmp
        self.Mirror(p.left)
        self.Mirror(p.right)

        return root
